source/common/options.js

```diff
@@ -149,12 +149,23 @@ function buildOptionsPage() {
 
   settings.forEach(function (setting) {
     if (setting.type === 'checkbox') {
-      jq('#' + setting.section + 'SettingsPage').append(jq('<div>', { class: 'row option-row' }).append(jq('<input>', { type: 'checkbox', id: setting.name, name: setting.name, 'aria-describedby': setting.name + 'HelpBlock' })).append(jq('<div>', { class: 'option-description' }).append(jq('<label>', { for: setting.name, text: setting.title })).append(jq('<span>', { id: setting.name + 'HelpBlock', class: 'help-block', text: setting.description }))));
+      jq('#' + setting.section + 'SettingsPage')
+        .append(jq('<div>', { class: 'row option-row' })
+          .append(jq('<input>', { type: 'checkbox', id: setting.name, name: setting.name, 'aria-describedby': setting.name + 'HelpBlock' }))
+          .append(jq('<div>', { class: 'option-description' })
+            .append(jq('<label>', { for: setting.name, text: setting.title }))
+            .append(jq('<span>', { id: setting.name + 'HelpBlock', class: 'help-block' }))));
+      jq('#' + setting.name + 'HelpBlock').html(setting.description);
```

Comment:
Last time we tried to do this we got rejected by Mozilla review. The reasoning went like this:

- This change looks innocuous enough, and right now your settings are all stored on disk.
- But if you ever changed your code to pull your settings in from a remote place, we wouldn't notice this because we'd only be reviewing diffs.
- And we'd then have allowed an extension with an XSS vulnerability in it.

I found that to be a bit of a stretch but that's what we were told last time and why it's currently text.

I reckon the best solution here is to allow Markdown in descriptions. This way it's not arbitrary html injection but will let you format whatever you'd like to format.